### PR TITLE
Migrate to smart pointers & following Google code style for C++

### PR DESCRIPTION
**Pull request by alexmercerind**
- Now following Google style guide for C++.
- Now using `std::shared_ptr` for all `MediaSource` instances everywhere, since same instance of `Media` or `Playlist` object can be opened again by user inside a `Player`, thus calling `std::move` everytime might not be best choice as it will delete object created by user (thus, can't be opened again unless created again) & object is needed copied within `Player` itself for handling lifecycle.
- Pretty much no memory leaks now.
- There is another PR coming for macOS support from @namniav, might cause some conflicts. I can't tell if it will work well with this & I don't have a mac to debug.

[August 13, 2021, 1:33 PM IST]

- Turns out everything else was fine. Just buffer allocation for `PlayerInternal::video_frame_buffer_` wasn't working.
- `std::make_unique<uint8_t>(size)` caused crash, switched to `video_frame_buffer_.reset(new uint8_t[size])` for now.

[August 13, 2021, 2:38 AM IST]

- Just like any other refactor ever done, it still has flaws & segmentation faults will occur. Will fix before the merge, will see the call-stack & breakpoints later on.
- Compile time exceptions shouldn't be there for now.

### CODE REVIEW

File: dartvlc/internal/events.hpp [hunk 1]
```diff
@@ -1,230 +1,223 @@
...
+
+  void OnPlaylist(std::function<void(void)> callback) {
+    playlist_callback_ = callback;
+  }
+
+  void OnVideo(std::function<void(uint8_t* frame)> callback) {
+    video_callback_ = callback;
+    int32_t pitch = video_width_ * 4;
+    int32_t size = video_height_ * pitch;
+    video_frame_buffer_.reset(new uint8_t[size]);
```
Comment (alexmercerind) [inline dartvlc/internal/events.hpp:74]: @jnschulze so, I have switched to smart pointers evevrywhere & following Google style guide for C++.

I tried to allocate buffer with `std::make_unique<uint8_t>(size)` but it wasn't working, so had to write this. I don't know if something better can be done.

Comment (jnschulze) [reply dartvlc/internal/events.hpp:74]: Basically, you can either write
`std::make_unique<uint8_t[]>(new uint8_t[size])`
or
`std::make_unique<uint8_t[]>(size)` which will initialize the memory

**Review by jnschulze — APPROVED** [review]
This looks like a huge step forward and is much easier to read now 🚀 
Let's get this merged.

File: ffi/native/dart_vlc.cpp [hunk 2]
```diff
@@ -25,374 +28,342 @@ extern "C" {
...
+  player->OnOpen([=](VLC::Media) -> void { OnOpen(id, player->state()); });
+  player->OnPlaylist([=]() -> void { OnOpen(id, player->state()); });
+#ifdef _WIN32
+/* Windows: Texture & flutter::TextureRegistrar */
+#else
+  /* Linux: decodeImageFromPixels & NativePorts */
+  if (player->video_width() > 0 && player->video_height() > 0) {
+    player->OnVideo([=](uint8_t* frame) -> void {
+      OnVideo(player->video_width() * player->video_height() * 4,
+              player->state(), frame);
```
Comment (namniav) [inline ffi/native/dart_vlc.cpp:67]: missing `id` argument

Comment (alexmercerind) [reply ffi/native/dart_vlc.cpp:67]: @namniav I couldn't notice it because I was building on Windows. Fixed on `master`.

**Pull request merged**